qiskit/aqua/operators/primitive_ops/matrix_op.py

```diff
@@ -218,5 +218,15 @@ def to_matrix_op(self, massive: bool = False) -> OperatorBase:
     def to_instruction(self) -> Instruction:
         return (self.coeff * self.primitive).to_instruction()  # type: ignore
 
+    def to_gate(self, parameter_map=None, label=None) -> Gate:
+        """ Returns a ``Gate`` equivalent to this Operator. """
+        if isinstance(self.primitive, List):  # pylint: disable=typecheck
```

Comment:
Can you explain what you did here? Shouldn't the primitive always be a matrix? Then you could return a `UnitaryGate` and pass the matrix as input.

Reply:
Ohh that's a much better way to do it. I'll try it using that method! Thanks 😄